ngageoint/opensphere
# Create config providers and add layer configs

Adds API's to:
* Create a config provider (`plugin.config.Provider.create`) and save it to settings.
* Add layers to a config provider (`plugin.config.Provider#addLayerGroup`).

Existing config provider/descriptor behavior should be unchanged. Existing config descriptor unit tests should cover old behavior, and additional tests have been added for new behavior.

Other changes:
* Most of `os.ui.file.ui.DefaultFileNodeUICtrl` has been abstracted out to `os.ui.data.DescriptorNodeUICtrl`. The former has been maintained to avoid a breaking change and for a minor adjustment to the removal warning.
* A few places were using `dataManager.getProvider(descriptor.getDescriptorType())` to locate a provider. This method requires the provider ID to match the descriptor type, which is not guaranteed to be true. This code has been cleaned up to use `descriptor.getId()` instead.
* Provider storage keys have been moved to an enum (`os.data.ProviderKey`) instead of using the strings directly.

Testing:
* Verify existing configuration with config providers functions as it did before.
* Configure a WMTS layer and verify it functions normally:
```
"wmts-test": {
  "type": "wmts",
  "label": "NASA GIBS WMTS",
  "url": "https://gibs.earthdata.nasa.gov/wmts/epsg4326/best/wmts.cgi"
}
```
* Build with https://github.com/wallw-bits/opensphere-plugin-geopackage and test loading GPKG files.
